webapp/survey/nodeDefs/reducer.js

```diff
@@ -23,12 +30,18 @@ const actionHandlers = {
   // Single nodeDef actions
   [nodeDefCreate]: (state, { nodeDef }) => NodeDefsState.assocNodeDef(nodeDef)(state),
 
-  [nodeDefPropsUpdate]: (state, { nodeDefUuid, props, propsAdvanced }) =>
-    NodeDefsState.assocNodeDefProps(nodeDefUuid, props, propsAdvanced)(state),
-
   [nodeDefDelete]: (state, { nodeDef }) => NodeDefsState.dissocNodeDef(nodeDef)(state),
 
+  [nodeDefUpdate]: (state, { nodeDef }) => NodeDefsState.assocNodeDef(nodeDef)(state),
+
   [nodeDefsUpdate]: (state, { nodeDefs }) => NodeDefsState.mergeNodeDefs(nodeDefs)(state),
+
+  [nodeDefPropsUpdateTemp]: (state, { nodeDef }) => NodeDefsState.assocNodeDef(nodeDef)(state),
+
+  [nodeDefPropsTempCancel]: (state, { nodeDef, nodeDefOriginal, isNodeDefNew }) =>
+    isNodeDefNew
+      ? NodeDefsState.dissocNodeDef(nodeDef)(state) // Remove node def from state
+      : NodeDefsState.assocNodeDef(nodeDefOriginal)(state), // Restore original version of node def
```

Comment:
why do you even change the nodeDef here when updating props? i suspect you don't need this at all

Reply:
survey node defs are used in several places, they must be up to date with the just applied changes, for example for the expression editor.

Reply:
that's true. good